File: bench/template/math/fact_rec/fact_rec.py
import json
import time

def fact(n):
    if n == 0:
        return 1
    return n * fact(n - 1)

n = {{ .N }}
repeat = 1000
last = 0

start = time.perf_counter()
for _ in range(repeat):
    last = fact(n)
duration = (time.perf_counter() - start) * 1000

print(json.dumps({
    "duration_ms": duration,
    "output": last,
}))

File: bench/template/math/fact_rec/fact_rec.ts
function fact(n: number): number {
    if (n === 0) return 1;
    return n * fact(n - 1);
}

const n = {{ .N }};
const repeat = 1000;
let last = 0;

const start = performance.now();
for (let i = 0; i < repeat; i++) {
    last = fact(n);
}
const duration = performance.now() - start;

console.log(JSON.stringify({
    duration_ms: duration,
    output: last,
}));
